Antidote/CallsManager.m

```diff
@@ -0,0 +1,331 @@
+{
+    self.manager.enableMicrophone = enableMicrophone;
+}
+
+- (BOOL)enableMicrophone
+{
+    return self.manager.enableMicrophone;
+}
+
+#pragma mark - ActiveCallViewController Delegate
+
+- (void)activeCallDeclineButtonPressed:(ActiveCallViewController *)controller
+{
+    [self.manager sendCallControl:OCTToxAVCallControlCancel toCall:self.currentCall error:nil];
+}
+
+- (void)activeCallMicButtonPressed:(ActiveCallViewController *)controller
+{
+    BOOL enable = controller.micSelected;
+    self.manager.enableMicrophone = enable;
+    controller.micSelected = ! enable;
+}
+
+- (void)activeCallSpeakerButtonPressed:(ActiveCallViewController *)controller
+{
+    NSError *error;
+
+    if (controller.speakerSelected) {
+        if ([self.manager sendCallControl:OCTToxAVCallControlUnmuteAudio toCall:self.currentCall error:&error]) {
+            controller.speakerSelected = NO;
+        }
+        else if (error.code == OCTToxAVErrorControlInvaldTransition) {
+            controller.speakerSelected = YES;
+        }
+    }
+    else {
+        if ([self.manager sendCallControl:OCTToxAVCallControlMuteAudio toCall:self.currentCall error:&error]) {
+            controller.speakerSelected = YES;
+        }
+        else if (error.code == OCTToxAVErrorControlInvaldTransition) {
+            controller.speakerSelected = NO;
+        }
+    }
+}
+
+- (void)activeCallDeclineIncomingCallButtonPressed:(ActiveCallViewController *)controller
+{
+    [self.manager sendCallControl:OCTToxAVCallControlCancel toCall:self.pendingIncomingCall error:nil];
+    self.pendingIncomingCall = nil;
+    controller.incomingCallCallerName = nil;
+    controller.createIncomingCallView = NO;
+}
+
+- (void)activeCallAnswerIncomingCallButtonPressed:(ActiveCallViewController *)controller
+{
+    [self.manager answerCall:self.pendingIncomingCall enableAudio:YES enableVideo:NO error:nil];
+    self.pendingIncomingCall = nil;
+    controller.incomingCallCallerName = nil;
+    controller.createIncomingCallView = NO;
+}
+
+- (void)activeCallPausedCallSelectedAtIndex:(NSUInteger)index controller:(ActiveCallViewController *)controller
+{
+    // grab paused call from RBQ index and resume call.
+}
+
+#pragma mark - DialingCallViewController Delegate
+
+- (void)dialingCallDeclineButtonPressed:(DialingCallViewController *)controller
+{
+    [self.manager sendCallControl:OCTToxAVCallControlCancel
+                           toCall:self.currentCall
+                            error:nil];
+}
+
+#pragma mark - RingingCallViewController Delegate
+
+- (void)ringingCallAnswerButtonPressed:(RingingCallViewController *)controller
+{
+    [self.manager answerCall:self.currentCall
+                 enableAudio:YES
+                 enableVideo:NO error:nil];
+}
+
+- (void)ringingCallDeclineButtonPressed:(RingingCallViewController *)controller
+{
+    [self.manager sendCallControl:OCTToxAVCallControlCancel
+                           toCall:self.currentCall
+                            error:nil];
```

Comment:
Yea it is a bit.  The reason is because in ToxAV they are two separate methods. There is no `sendCallControlAnswer` for answering calls. Which would of been nice.